packages/framework/src/types/util.types.ts

```diff
@@ -1,3 +1,8 @@
+/*
+ * THIS FILE SHOULD NOT DEPEND ON ANY OTHER FILES.
+ * IT SHOULD ONLY CONTAIN UTILITY TYPES.
+ */
+
```

Comment:
Just a simple warning message to ensure we don't get create any circular deps here!  😀